qiskit/providers/ibmq/runtime/runtime_program.py

```diff
@@ -255,3 +258,77 @@ class ProgramResult(NamedTuple):
     name: str
     description: str
     type: str
+
+class ParameterNamespace(SimpleNamespace):
+    """ An abstraction for SimpleNamespace that offers param validation.
+
+    Note: Unused/Unnecessary params does not cause error (still validates)
+
+    Args:
+        params (List[ProgramParameter]): The program's input parameters
+
+    Raises
+        IBMQInputValueException on validation fail
+    """
+
+    def __init__(self, params: List[ProgramParameter]):
+        super().__init__()
+        # Allow access to the raw program parameters list
+        self.params_list = params
```

Comment:
1) Can we give it a better name than `param_list`? I for one find `program.parameter.params_list` quite confusing. How about `program_metadata` or just `metadata`? Since that's the term we use for the "documentation" part of a program.
2) I usually try to make attributes that should not be modified by users private and readable via a property. This prevents users from accidentally overwrite it.

Reply:
Fixed!